### DISCUSSION TITLE
Module / Plugins infrastructure

### DISCUSSION THREAD

**Discussion by TrungRueta** [remix-run/remix#2180] [Proposals]
Hi Remix team,

First, i would like to say congraturation Remix reached to v1. This is wonderful framework and i believe FW will have more and more features in future and popular !!

I would like to request we have good document about how to write module / plugin as third-party for remix. For now alot libraries can not co-op with Remix (for example Windicss / Tailwindcss still must use hack way with double CLI commands to keep running). 

By expose module / plugins apis allow community can join in and write plugin serve app. 
Hope to hear ideas from everyone

Thanks again !
Trung

**Comment by MichaelDeBoey** [comment]
As @kentcdodds mentioned in https://github.com/remix-run/remix/pull/2168#issuecomment-1058193715, it's unlikely to happen that the Remix-team will expose the `esbuild` config.

> [...] This is extremely unlikely to happen. I'm not sure whether @ryanflorence or @mjackson have written out why we don't expose the build config anywhere (I think they have, but can't remember where), but I can say briefly that by exposing the build config it makes it too easy for people to shoot themselves in the foot and limits our ability to make changes that don't break people's apps.
> 
> We're much more interested in helping address specific use cases that people have and building in solutions for common use cases to building web apps.
> 
> For the edge cases, I know it's probably not what you want to hear, but in the early days of Remix, most of the Remix community made use of https://npm.im/patch-package to get features that Remix hadn't yet implemented and it actually worked out great. It's not something we'd necessarily recommend, but it's a pretty solid solution for folks who need something that we're planning on supporting in the future but isn't ready yet.
